### ISSUE TITLE
Forum data migration

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#1518]
Goal: discuss the prospects and purpose of a forum data migration

**Comment by DanVanAtta** [comment]
My thoughts on this subject are not to do it, just copy individual thresds over as desired. A bulk migration is really hard to justify.
- The old forums (SF and warclub) are still available. That will be the case for some time. Eventually it'll be natural to turn them read-only and/or archive them. All the while we can still reference and search them.
- I don't know if adding inactive conversation threads to a new forum is actually useful.. Any reference information in the 'old' threads are better summarized, updated, and posted with the documentation.
- If we can do it a bulk migration now, we can do it at a future date as well. So there is no pressing need to merge that data in now. We can see how things play out a bit and re-evaluate.
- Technically it may be really challenging, perhaps not even possible to do a bulk data migration. It's very hard to justify such a massive effort when instead we can sink that effort into another area. We might be talking a several month project. We really can use that effort better elsewhere. Ideally we'd farm out the task of scanning for stuff that is particularly useful/current, and summarizing it and putting it in with the game help documentation.

**Comment by beelee1** [comment]
@DanVanAtta  I guess I should answer here about copying conversations ? No, I just wanted to be able to search for answers so I don't have to bug people as often.  Sounds as if that might not be possible at the Triplea Dev Forum if there's no read only ? 

Not that big a deal Imo. If it can't be done or isn't worth it time/compatibility wise, then a fresh start it is.

**Comment by prastle** [comment]
also when i copy a few over as test.  some work some dont.  a few things i will just post

 but yes copy paste is a huge job

**Comment by prastle** [comment]
Realistically it would be nice 1 forum 1 place for all and if they want old data they bring themselves


jmho Pras

**Comment by DanVanAtta** [comment]
I think there is agreement that any content folks want moved over, they can copy/paste it. No plans for any kind of bulk data migration.

**Comment by RoiEXLab** [comment]
Agreed @DanVanAtta

**Comment by Cernelius** [comment]
As I said, and just for the record, my uninformed opinion is that the best (not saying feasible) would be to move all to the new forum, and then fully delete both old ones, instead of keeping 3 forums, of which 2 frozen. But nevermind, if the developers think that is not good, for tech reasons.

**Comment by RoiEXLab** [comment]
@Cernelius I get your point, I see why it would be useful to have all data move to the new forum.
There is just a Problem with this... Although it wouldn't be impossible to copy all the forum posts (inserting them into NodeBB would actually be super easy), it would be very hard to retrieve all the forum data via HTTP... There would be a way to iterate through the xoops database, but therefore we'd need to figure out how this database works as well. Also what account should this post be posted as? As @DanVanAtta pointed out, we can't migrate user accounts due to the way passwords are stored (probably bycrypt hashing-algo).
Also as I mentioned earlier I don't think it would be ideal to migrate every unimportant topic over.

TL;DR it would be possible to migrate all data in some way, but IMO not very viable and useful

**Comment by Cernelius** [comment]
I was actually mostly thinking about importing the Developer forum, rather than the WarClub, about the fact that it would more neat if it gets imported, and then deleted, instead of freezed forever. I think the WarClub can be pretty much just deleted with minor arm, kind of, aside relatively minor issue to maybe losing the tournament history (but I don't think people read it). I don't care about Ladder, but I'm thinking about the Ladder rules? Like where it says you can't bid a Bomber in v3 etc.? Those disappeared when the Ladder broke down.
Sometimes it happens that forums move entirely from one to another, so I understood this is what it was gona happen. I guess moving up the SourceForge database is not feasible either.
Again, whatever, not strong on it, nor need, you guys go ahead as you believe, of course.

**Comment by prastle** [comment]
Valid point we need to rewrite the ladder rules
even if no ladder

**Comment by Cernelius** [comment]
Ok, I found where I understood that merging meant merging:
https://github.com/triplea-game/triplea/issues/1071
@ron-murhammer 
> I'd say we are 90% confirmed on this. We need to investigate how to technically do the migration as well. The SF forum uses nabble and the warclub forum uses newbb. My guess is that we will need to export the SF forum to something like XML then write some scripts to feed the data into the warclub forum (I doubt there is any kind of converter between them).

So, I understood that all was decided to be copy pasted into WarClub first, and then all moved to new. Getting away with the WarClub intermediate step, I just assumed all would have been just moved there directly.
Having no clue myself, I guess it was a reasonable assumption from me, based on the discussions here and the ones that were opened in WarClub as well, but I see now the matter was not decided, but actually discussed between merging and just freezing; I totally overlooked that.
So, again, just pointing it out, since it seemed @ron-murhammer  had in mind a way to do it (probably by now sorted out not feasible).

**Comment by DanVanAtta** [comment]
Closing, bulk data migration is not happening.
